src/context.hpp

```diff
@@ -120,6 +126,8 @@ struct cvk_context : public _cl_context,
     size_t m_printf_buffersize;
     cvk_printf_callback_t m_printf_callback;
     void* m_user_data;
+
+    cvk_command_queue* m_queue_image_init = nullptr;
```

Comment:
`std:unique_ptr`? You wouldn't need the free function.

Reply:
I agree, that's what I tried to do initially. But it is not simple. clvk's headers contain too much code everywhere leading to build errors (you can see that `context.hpp` does not include `queue.hpp` in this PR, it's only declaring `struct cvk_command_queue`, thus making the destructor inaccessible for `std::unique_ptr`).

I've tried to move a few things around, but it quickly started to be too much making this PR too complicated for what it is trying to achieve.

I can open an issue to have a rework task for it to do once this land if you want.

Reply:
Right, we do need to untangle those headers a bit. I've already bumped into similar issues in the past. Please create an issue with as many specifics as possible. Thanks!

Reply:
https://github.com/kpet/clvk/issues/735